bazel/ray_deps_setup.bzl

```diff
@@ -1,128 +1,198 @@
 load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository", "new_git_repository")
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-def ray_deps_setup():
-    RULES_JVM_EXTERNAL_TAG = "2.10"
-    RULES_JVM_EXTERNAL_SHA = "1bbf2e48d07686707dd85357e9a94da775e1dbd7c464272b3664283c9c716d26"
+def github_repository(**kwargs):
+    """
+    Conveniently chooses between ZIP, git, etc. GitHub repositories.
+    We prefer ZIP downloads, since they're smaller and faster
```

Comment:
```suggestion
    Recommend to download zip packages, because zip packages have no history of the repositories.
```

Reply:
BTW, the comments of this method seems too verbose, could you refine them?

Reply:
I agree it's more verbose for my liking too, but I've in fact already rewritten it several times to get it into this shape. :) I removed the parameters that are already described in Bazel and only documented any additional knowledge needed. I'm not sure what else to condense without removing pertinent information to be honest. I'm happy to consider an alternative if you have an suggestions on how to shorten it, though.